osc/gitea_api/repo.py

```diff
@@ -0,0 +1,67 @@
+from typing import Optional
+
+from .connection import Connection
+from .exceptions import BranchExists
+from .exceptions import ForkExists
+from .exceptions import GiteaException
+
+
+def branch(
+    conn: Connection,
+    owner: str,
+    repo: str,
+    *,
+    new_branch_name: str,
+    old_ref_name: Optional[str] = None,
+):
+    """
+    Create a new branch in a repository.
```

Comment:
I would find it more useful, if this function would ensured that a branch is there. `ensure_branch` is not a great name, but something along those lines. This essentially requires that I have my own wrapper for this function which does:
```
try:
    branch(c, user, repo)
except BranchExists:
    pass
```

Reply:
How about to follow `os.makedirs()` and introduce `exist_ok = False` argument?

Reply:
yes that would be perfect